### PR TITLE
Use sdk model for pipeline input

### PR DESCRIPTION
**Pull request by olaughter**
The sdk & backend both had the same identical model. This makes changes more convoluted and runs the risk of growing divergence over time. Here we consolidate them to use the sdk version instead, meaning that new fields can be managed in one place.

Note that there was the small config difference already which I have preserved by inheriting the sdk model into the existing backend model. I wanted to minimise how much I was changing all in one go but we can take this further in the future if we would like to.

See [all sdk changes](https://github.com/climatepolicyradar/cpr-sdk/compare/v1.1.9...v1.3.11) that are introduced here. (Essentially various version bumps + the model changes). And the [code for the sdk model](https://github.com/climatepolicyradar/cpr-sdk/blob/main/src/cpr_sdk/pipeline_general_models.py#L16) replacing the backend model.

## Proposed version

Please select the option below that is most relevant from the list below. This
will be used to generate the next tag version name during auto-tagging.

- [ ] Skip auto-tagging
- [x] Patch
- [ ] Minor version
- [ ] Major version

Visit the [Semver website](https://semver.org/#summary) to understand the
difference between `MAJOR`, `MINOR`, and `PATCH` versions.

Notes:

- If none of these options are selected, auto-tagging will fail
- Where multiple options are selected, the most senior option ticked will be
  used -- e.g. Major > Minor > Patch
- If you are selecting the version in the list above using the textbox, make
  sure your selected option is marked `[x]` with no spaces in between the
  brackets and the `x`

## Type of change

Please select the option(s) below that are most relevant:

- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change
- [ ] GitHub workflow update
- [ ] Documentation update
- [x] Refactor legacy code
- [ ] Dependency update

## How Has This Been Tested?

Please describe the tests that you added to verify your changes.

## Reviewer Checklist

- [ ] The PR represents a single feature (small driveby fixes are also ok)
- [ ] The PR includes tests that are sufficient for the level of risk
- [ ] The code is sufficiently commented, particularly in hard-to-understand areas
- [ ] Any required documentation updates have been made
- [ ] Any TODOs added are captured in future tickets
- [ ] No FIXMEs remain

### CODE REVIEW

**Review by katybaulch — APPROVED** [review]
ughhh beautiful tidy, have all the stars ⭐ 🌟

File: tests/unit/app/core/test_pipeline.py [hunk 1]
```diff
@@ -12,10 +12,9 @@ def test_generate_pipeline_ingest_input(data_db: Session):
 
     documents = generate_pipeline_ingest_input(data_db)
     assert len(documents) == 2
-    assert documents[0].name == "Fam1"
-    assert documents[0].import_id == "CCLW.executive.2.2"
-    assert documents[1].name == "Fam1"
-    assert documents[1].import_id == "CCLW.executive.1.2"
+    for d in documents:
+        assert d.name == "Fam1"
+        assert d.import_id in ["CCLW.executive.2.2", "CCLW.executive.1.2"]
```
Comment (diversemix) [inline tests/unit/app/core/test_pipeline.py:17]: The logic here is wrong - as it would pass if it returned one of these documents twice

Comment (diversemix) [reply tests/unit/app/core/test_pipeline.py:17]: (I corrected the logic in the PR that just got merged)

File: app/api/api_v1/schemas/document.py [hunk 2]
```diff
@@ -1,6 +1,7 @@
 from datetime import datetime
 from typing import Any, Mapping, Optional, Sequence, Union
 
+from cpr_sdk.pipeline_general_models import BackendDocument
```
[the remark this answers is not in the document]
Comment (olaughter) [reply app/api/api_v1/schemas/document.py:4]: Adding hte details mentioned on slack here: Yep, the pipeline uses the same model and yes the Sdk model is already updated ready for you

File: tests/unit/app/core/test_pipeline.py (hunk 1 above)
Comment (olaughter) [reply tests/unit/app/core/test_pipeline.py:17]: Great spot, thanks Peter!

**Pull request merged**